### PR TITLE
Add `CenterPoint` and `Bounds` methods

### LINKED ISSUE #1386: Align viewports in outline
#### Description:

Add methods needed for aligning existing viewports in given outline / drawing area of sheet - it is needed for further development of `Align Views on Sheet` feature.

### PR DESCRIPTION
**Pull request by michal-pekacki**
### Issues addressed by this PR

Closes #1386 

- `Bounds` of the outlines method added
- `CenterPoint` of the outline method added

### CODE REVIEW

File: Revit_oM/Enums/OutlineAlignment.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+using System.ComponentModel;
+
+namespace BH.oM.Revit.Enums
+{
+    /***************************************************/
+
+    [Description("Direction of the sheet viewport alignment.")]
+    public enum OutlineAlignment
+    {
```
Comment (pawelbaran) [inline Revit_oM/Enums/OutlineAlignment.cs:31]: Do you think we will ever have other distribution layouts? To me an enum feels a bit too constrained, I would consider an interface instead, for example:
- `IViewDistribution` interface (no properties? or maybe `Gap`)
- `DistributionInRows` class implementing `IViewDistribution` with properties that would define left to right, top to down etc.
- `DistributionInColumns` class implementing `IViewDistribution` with properties that would define left to right, top to down etc.
- distribution method would take `IViewDistribution`

...this is just an idea off the top of my head, so not sure if valid. Thinking out loud.

File: Revit_Core_Engine/Query/AlignOutlines.cs [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+    public static partial class Query
+    {
+        /***************************************************/
+        /****               Public methods              ****/
+        /***************************************************/
+
+        [Description("Aligns outlines in the bounding outline along a given direction.")]
+        [Input("boudingOutline", "Bounding outline in which outline will be aligned.")]
+        [Input("outlines", "Outlines to aligned in the bounding outline.")]
+        [Input("offset", "Distance between each outline.")]
```
[the remark this answers is not in the document]
Comment (michal-pekacki) [reply Revit_Core_Engine/Query/AlignOutlines.cs:41]: This method has been deleted from this PR.

File: Revit_Core_Engine/Query/AlignOutlines.cs [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+        [Input("direction", "Direction of the outline alignment.")]
+        [Output("alignedOutlines", "Outlines aligned in the bounding outline.")]
+        public static List<Outline> AlignOutlines(this Outline drawingArea, List<Outline> outlines, double offset, OutlineAlignment direction)
```
[the remark this answers is not in the document]
Comment (michal-pekacki) [reply Revit_Core_Engine/Query/AlignOutlines.cs:44]: This method has been deleted from this PR.

File: Revit_Core_Engine/Query/Bounds.cs [hunk 4]
```diff
@@ -81,6 +81,27 @@ public static BoundingBoxXYZ Bounds(this IEnumerable<Solid> solids, Transform tr
...
 
         /***************************************************/
+
+        [Description("Returns the combined outline of the given outlines.")]
+        [Input("outlines", "A list of outlines to combine.")]
+        [Output("bounds", "Combined outline of the given outlines.")]
+        public static Outline Bounds(this List<Outline> outlines)
+        {
+            if (outlines == null || !outlines.Any())
+                return null;
```
Comment (pawelbaran) [inline Revit_Core_Engine/Query/Bounds.cs:91]: I would throw an error about null or empty collection here

**Review by pawelbaran — COMMENTED** [review]
One last bit from my side 👍

File: Revit_Core_Engine/Query/Bounds.cs (hunk 4 above)
Comment (michal-pekacki) [reply Revit_Core_Engine/Query/Bounds.cs:91]: Done 👍

**Comment by michal-pekacki** [comment]
@BHoMBot check compliance

**Review by pawelbaran — APPROVED** [review]
The code makes full sense to me now, thanks @michal-pekacki 👍

**Comment by michal-pekacki** [comment]
@BHoMBot check required

**Pull request merged**